app/controllers/users_controller.rb

```diff
@@ -38,22 +40,33 @@ def get_user
   end
 
   def index
+    if params[:sort]
+      @had_skill_skill_id = params[:sort][:had_skill_skill_id].to_i
+      @wanted_skill_skill_id = params[:sort][:wanted_skill_skill_id].to_i
+    end
     if params[:query]
-      @users = User.search(params[:query]).limit(6).page(params[:page])
+      @users = User.search(params[:query])
+    elsif @had_skill_skill_id != nil && @wanted_skill_skill_id != nil
+      @users = find_matching_had_wanted(@had_skill_skill_id, @wanted_skill_skill_id)
+      respond_to do |format|
+        format.html # index.html.erb
+        format.json { render json: @articles }
+        format.js
+      end
+    elsif params[:miles] == 1
+      @user = nearby_users
+      respond_to do |format|
+        format.html # index.html.erb
+        format.json { render json: @articles }
+        format.js
+      end
+    elsif params[:miles] == 5
+    elsif params[:miles] == 25
     else
-      @users = User.order('created_at DESC').limit(6).page(params[:page])
+      @users = User.order('created_at DESC')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.